app/controllers/advertisements_controller.rb

```diff
@@ -25,14 +26,21 @@ def ip_info
     @ip_info ||= MMDB.lookup(request.remote_ip)
   end
 
+  def property_id
+    @property_id ||= begin
+      return Property.where(legacy_id: params[:legacy_id]).pluck(:id).first if params[:legacy_id].present?
+      params[:property_id].to_i
+    end
+  end
```

Comment:
Does the early return here work as expected? i.e. memoize by assigning `@property_id` or does it just return from the method?